streams/src/test/java/org/apache/kafka/streams/integration/InternalTopicIntegrationTest.java

```diff
@@ -113,23 +113,29 @@ private void produceData(final List<String> inputValues) throws Exception {
     }
 
     private Properties getTopicProperties(final String changelog) {
-        try (KafkaZkClient kafkaZkClient = KafkaZkClient.apply(CLUSTER.zKConnectString(), false,
-                DEFAULT_ZK_SESSION_TIMEOUT_MS, DEFAULT_ZK_CONNECTION_TIMEOUT_MS, Integer.MAX_VALUE,
-                Time.SYSTEM, "testMetricGroup", "testMetricType")) {
-            final AdminZkClient adminZkClient = new AdminZkClient(kafkaZkClient);
-            final Map<String, Properties> topicConfigs =
-                JavaConverters.mapAsJavaMapConverter(adminZkClient.getAllTopicConfigs()).asJava();
-
-            for (Map.Entry<String, Properties> topicConfig : topicConfigs.entrySet()) {
-                if (topicConfig.getKey().equals(changelog)) {
-                    return topicConfig.getValue();
+        try (final AdminClient adminClient = createAdminClient()) {
+            final ConfigResource configResource = new ConfigResource(ConfigResource.Type.TOPIC, changelog);
+            try {
+                final Config config =  adminClient.describeConfigs(Collections.singletonList(configResource)).values().get(configResource).get();
```

Comment:
nit: double space after `=`

Reply:
Similarly, `adminClient.describeConfigs` does not read from ZK but from broker cache, and hence maybe subject to race conditions.

Reply:
hmm..not sure how handle this?  Can we add small Thread.sleep to method? or some kind of waitForCondition implementation?

Reply:
Thinking about this a bit more, using `adminClient.describeConfigs` here would be okay since we never dynamically change the config, we just want to test that the configs specified when the topic was created is expected. So if the topic creation is not yet propagated, it will fail and re-try on admin client anyways. So nvm.